### ISSUE TITLE
All-Inkl - sha1 is deprecated and will be discontinued in December 2022

### ISSUE THREAD

**Issue by webprofusion-chrisc** [rmbolger/Posh-ACME#459]
Currently the All-Inkl provider will generally use sha1 even for "plain" passwords:
https://github.com/rmbolger/Posh-ACME/blob/main/Posh-ACME/Plugins/All-Inkl.ps1#L213

The following message (edited by me to remove customer id) was received from All-Inkl support via a Certify The Web user who was utilising the Posh-ACME All-Inkl provider. I don't have a test account to verify this with and I can't see anything announcing the sha1 deprecation on their website but I do see that the lego client also recently removed sha1 for All-Inkl (https://github.com/go-acme/lego/pull/1713).

The solution would seem to be to completely remove sha1 support from the All-Inkl provider and just use plain text?

> Our system has detected that you are using the KAS API, which means that your custom scripts gain access to several features of the technical administration panel (KAS). So please forward the following information to the developer of those custom scripts as appropriate.
> 
> The authentication method "sha1" which is used by your custom scripts is deprecated and support for this authentication method will be gradually discontinued starting December 2022. 
> 
> Here's an example of how such a script code currently looks like: 
> 
> $auth=array(
>  'kas_login' => 'w12345',
>  'kas_auth_type' => 'sha1',
>  'kas_auth_data' => sha1('myPassword')
> )
> 
> Please alter your scripts in a way so that they use the new authentication method "plain" instead. An example how it should look afterwards: 
> 
> $auth=array(>  'kas_login' => 'w12345',
>  'kas_auth_type' => 'plain',
>  'kas_auth_data' => 'myPassword'
> )

> The complete documentation of the KAS API documentation can also be found again at http://kasapi.kasserver.com/dokumentation/phpdoc/

**Comment by rmbolger** [comment]
Man, what a weird situation. I wonder if All-Inkl is in the process of migrating to sha256, but for some reason needs to remove sha1 before that happens. The original plugin contributor (#335) allowing for plain text passwords but then hashing them under the hood when it's not necessary is odd, but well intentioned.

Fixing the plugin so that passing the plaintext password will actually send it through to the API as-is shouldn't be terribly difficult. But I'm also stuck with no way to test it. All-Inkl does apparently provide API testing accounts, but it turns out they can't make DNS changes so for our purposes, they're worthless. When the plugin was originally submitted, the contributor was able to setup some sort of sub-account for me on his own account that I could test with.

I'll work on the fix and would your user be able to test it?

**Comment by rmbolger** [comment]
Here's the raw updated plugin file.

https://raw.githubusercontent.com/rmbolger/Posh-ACME/all-inkl-fix/Posh-ACME/Plugins/All-Inkl.ps1

**Comment by webprofusion-chrisc** [comment]
Thanks! I'll send them the file and see if they are willing to test :)

**Comment by rmbolger** [comment]
Hey @webprofusion-chrisc, any luck on getting the change tested? Hoping to hold off on the next release until this is merged.

**Comment by webprofusion-chrisc** [comment]
No sorry, user emails stopped bouncing but no reply.

**Comment by astaerk** [comment]
Hi, yes I'm still using it and I just realized that I also received this mail.

Back then I hashed plain passwords under the hood because there was some info on their page or in a support answer that I interpreted as plain passwords probably going away in the near future. Obviously it is the other way around now 🙄

This month I'm doing a lot of overtime for my primary job. But I will test this in the next days. Over the weekend by the latest
